### PR TITLE
[MIRROR] reworks emote sound code & adds screaming (Which only works if the code forces it. *scream wont make a sound)

### PR DESCRIPTION
**Pull request by ghost**
Original PR: https://github.com/tgstation/tgstation/pull/42216
----------------------------
:cl: Floyd / Qustinnus
soundadd: involuntary screaming for when you get hurt.
code: reworks emote sound handling to not use fucking s u b t y p i n g
/:cl:

so basically I like sounds and screaming adds some important feelies so I made it so that if the code forces you to emote a scream, it will make you scream audibly with audio. but *scream doesnt work (to prevent people ruining it like they did with every other fun feature)

https://streamable.com/yfksw

### CODE REVIEW

**Review by Carbonhell — DISMISSED** [review]

**Review by Carbonhell — COMMENTED** [review]
i need to check if we need to rework any of our existing code, ignore my approval

**Review by chesse20 — DISMISSED** [review]
Looks Good Hit that Merge Button

**Comment by Manly-Dwarf** [comment]
You should probably disable all of this.

File: code/modules/mob/living/carbon/human/emote.dm [hunk 1]
```diff
@@ -45,6 +45,31 @@
 	message = "mumbles!"
 	emote_type = EMOTE_AUDIBLE
 
+/datum/emote/living/carbon/human/scream
+	key = "scream"
+	key_third_person = "screams"
+	message = "screams!"
+	emote_type = EMOTE_AUDIBLE
+	only_forced_audio = TRUE
```
Comment (GuyonBroadway) [inline code/modules/mob/living/carbon/human/emote.dm:54]: Ok so if I have this right all we need to do is mirror this datum overriding "only_forced_audio" to false.

File: code/modules/mob/living/carbon/human/emote.dm [hunk 2]
```diff
@@ -45,6 +45,31 @@
...
 
+/datum/emote/living/carbon/human/scream
+	key = "scream"
+	key_third_person = "screams"
+	message = "screams!"
+	emote_type = EMOTE_AUDIBLE
+	only_forced_audio = TRUE
+	vary = TRUE
+
+/datum/emote/living/carbon/human/scream/get_sound(mob/living/user)
```
Comment (GuyonBroadway) [inline code/modules/mob/living/carbon/human/emote.dm:57]: we mirror THIS proc in place of https://github.com/HippieStation/HippieStation/blob/18cb55cb3e31c06a51a44736583876c77cf77a54/hippiestation/code/modules/mob/living/emote.dm#L4

and just account for all our species in it.

File: code/modules/mob/living/carbon/human/emote.dm [hunk 3]
```diff
@@ -45,6 +45,31 @@
...
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/H = user
+	if(!H.mind?.miming)
+		return
```
Comment (GuyonBroadway) [inline code/modules/mob/living/carbon/human/emote.dm:62]: hereish we put
`		LAZYINITLIST(user.alternate_screams)
		if(LAZYLEN(user.alternate_screams))
return pick(user.alternate_screams)`

File: hippiestation/code/modules/mob/living/emote.dm [hunk 4]
```diff
@@ -1,102 +1,104 @@
...
+	if(LAZYLEN(user.alternate_screams))
+		return pick(user.alternate_screams)
+	if(ismonkey(user))
+		return 'hippiestation/sound/voice/scream_monkey.ogg'
+	if(istype(user, /mob/living/simple_animal/hostile/gorilla))
+		return 'sound/creatures/gorilla.ogg'
+	if(ishuman(user))
+		user.adjustOxyLoss(user.scream_oxyloss)
+		return pick('hippiestation/sound/voice/scream_m1.ogg', 'hippiestation/sound/voice/scream_m2.ogg')
+		if(user.gender == FEMALE)
```
Comment (GuyonBroadway) [inline hippiestation/code/modules/mob/living/emote.dm]: Poot gendercheck last else catbeasts will sound like humans if female.

**Review by GuyonBroadway — DISMISSED** [review]

**Comment by Jujumatic** [comment]
Is this ok to merge now?

**Review by Jujumatic — APPROVED** [review]

**Pull request merged**